src/cpp/core/r_util/RActiveSessionStorage.cpp

```diff
@@ -26,153 +25,165 @@ namespace r_util {
+
+      if(iter != propertyValue.end())
+         *pValue = iter->second;
+      
+      return Success();
+   }
+
+   Error FileActiveSessionStorage::readProperties(const std::set<std::string>& names, std::map<std::string, std::string>* pValues)
+   {
+      std::vector<FilePath> failedFiles{};
+      pValues->empty();
+      for (const std::string &name : names)
+      {
+         FilePath readPath = getPropertyFile(name);
+         std::string value = "";
+
+         if (readPath.exists())
+         {
+            Error error = core::readStringFromFile(readPath, &value);
             if (error)
-                failedFiles.push_back(writePath);
-        }
-        
-        if (failedFiles.empty())
-            return Success();
-        else
-            return createError("UnableToWriteFiles", "Failed to write to the following files ", 
-                failedFiles, ERROR_LOCATION);
-    }
-
-    FilePath FileActiveSessionStorage::getPropertyDir() const
-    {
-        FilePath propertiesDir = scratchPath_.completeChildPath(propertiesDirName_);
-        propertiesDir.ensureDirectory();
-        return propertiesDir;
-    }
-
-    FilePath FileActiveSessionStorage::getPropertyFile(const std::string& name) const
-    {
-        FilePath propertiesDir = getPropertyDir();
-        const std::string& fileName = getPropertyFileName(name);
-        return propertiesDir.completeChildPath(fileName);
-    }
-
-    std::shared_ptr<IActiveSessionStorage> ActiveSessionStorageFactory::getFileActiveSessionStorage(const FilePath& scratchPath)
-    {
-        return std::make_shared<FileActiveSessionStorage>(FileActiveSessionStorage(scratchPath));
-    }
+               failedFiles.push_back(readPath);
+            boost::algorithm::trim(value);
+         }
+         pValues->insert(std::pair<std::string, std::string>{name, value});
+      }
+
+      if(failedFiles.empty())
+         return Success();
+      else
+         return createError("UnableToReadFiles", "Failed to read from the following files ", 
+            failedFiles, ERROR_LOCATION);
+   }
+
+   Error FileActiveSessionStorage::readProperties(std::map<std::string, std::string>* pValues)
+   {
+      FilePath propertyDir = getPropertyDir();
+      std::vector<FilePath> files{};
+      std::vector<FilePath> failedFiles{};
+      pValues->empty();
+      propertyDir.getChildren(files);
+
+      for(FilePath file : files) {
+         std::string value = "";
+         Error error = core::readStringFromFile(file, &value);
+
+         if(error)
+            failedFiles.push_back(file);
+
+         std::string propertyName = getFileNameProperty(file.getFilename());
+         pValues->insert(std::pair<std::string, std::string>{propertyName, value});
+      }
+
+      if(failedFiles.empty())
+         return Success();
+      else
+         return createError("UnableToReadFiles", "Failed to read from the following files ",
+            failedFiles, ERROR_LOCATION);
```

Comment:
Nit: prefer an unqualified `return Success();` at the end
```suggestion
      if(!failedFiles.empty())
         return createError("UnableToReadFiles", "Failed to read from the following files ",
            failedFiles, ERROR_LOCATION);
            
      return Success();
```